infra-examples/aws/eks/eks.tf

```diff
@@ -0,0 +1,178 @@
+      from_port   = 0
+      to_port     = 0
+      type        = "ingress"
+      self        = true
+    }
+    cluster_to_node = {
+      description                   = "Cluster to ingress-nginx webhook"
+      protocol                      = "-1"
+      from_port                     = 8443
+      to_port                       = 8443
+      type                          = "ingress"
+      source_cluster_security_group = true
+    }
+  }
+
+  # Disable secrets encryption
+  cluster_encryption_config = {}
+
+  iam_role_use_name_prefix = var.iam_role_use_name_prefix
+  iam_role_name            = var.iam_role_name
+
+  # for security group
+  cluster_security_group_description     = var.cluster_security_group_description
+  cluster_security_group_use_name_prefix = var.cluster_security_group_use_name_prefix
+  cluster_security_group_name            = var.cluster_security_group_name
+
+  eks_managed_node_group_defaults = {
+    iam_role_additional_policies = { for s in local.node_group_defaults_iam_role_additional_policies : s.key => s.value if s.enable }
+  }
+
+  eks_managed_node_groups = {
+    ubuntu_worker = {
+
+      ami_id     = var.ami_id != "" ? var.ami_id : data.aws_ami.latest_ubuntu_eks.id
+      key_name   = var.key_name
+      name       = var.node_group_name
+      subnet_ids = coalesce(var.node_group_subnets, module.vpc.private_subnets)
+
+      # This will ensure the boostrap user data is used to join the node
+      # By default, EKS managed node groups will not append bootstrap script;
+      # this adds it back in using the default template provided by the module
+      # Note: this assumes the AMI provided is an EKS optimized AMI derivative
+      enable_bootstrap_user_data = true
+
+      instance_types = var.instance_types
+      max_size       = var.max_size
+      min_size       = var.min_size
+      desired_size   = var.desired_size
+      capacity_type  = var.capacity_type
+
+      create_security_group = false
+
+      post_bootstrap_user_data = local.post_bootstrap_user_data
+
+      block_device_mappings = {
+        sda1 = {
+          device_name = "/dev/sda1"
+          ebs = {
+            volume_size = var.disk_size
+          }
+        }
+      }
+
+      tags = merge(var.node_groups_tags, local.cluster_autoscaler_tags)
+    }
+  }
+}
+
+module "ebs_csi_irsa_role" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 5.5.7"
+
+  role_name             = "ebs-csi-controller-${var.cluster_name}"
+  attach_ebs_csi_policy = true
+  tags                  = var.tags
+
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["kube-system:ebs-csi-controller-sa"]
+    }
+  }
+}
+
+# Role required by cluster_autoscaler
+module "cluster_autoscaler_irsa_role" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 5.5.7"
```

Comment:
Could we use a newer version of the [module](https://github.com/terraform-aws-modules/terraform-aws-iam)? This constraint only allows installing newer patches, keeping the same minor